Add modifier (#1209)
Codecademy/docs: content/swiftui/concepts/modifiers/modifiers.md

---

> Title: 'Modifiers'

Reviewer: ```suggestion
Title: 'ViewModifier'
```

> Description: 'Modifiers are methods that can alter views.'

Reviewer: ```suggestion
Description: 'ViewModifiers are methods that can alter views.'
```

Subjects:
 - 'Mobile Development'
 - 'Computer Science'
Tags:
 - 'SwiftUI'
 - 'SwiftUI Modifiers'
CatalogContent:
 - 'learn-swift'
 - 'paths/build-ios-apps-with-swiftui'
---

> In [SwiftUI](https://www.codecademy.com/resources/docs/swiftui), **Modifiers** are methods that can be called on a particular [View](https://www.codecademy.com/resources/docs/swiftui/views).

Reviewer 2: ```suggestion
In [SwiftUI](https://www.codecademy.com/resources/docs/swiftui), **modifiers** are methods that can be called on a particular [View](https://www.codecademy.com/resources/docs/swiftui/views).
```

Reviewer: I updated this to display the correct term to use here,

> In [SwiftUI](https://www.codecademy.com/resources/docs/swiftui), **Modifiers** are methods that can be called on a particular [Views](https://www.codecademy.com/resources/docs/swiftui/views).

Reviewer 3: ```suggestion
In [SwiftUI](https://www.codecademy.com/resources/docs/swiftui), **Modifiers** are methods that can be called on a particular [View](https://www.codecademy.com/resources/docs/swiftui/views).
```

## Syntax

> A `modifier` is a method that when called on a `View`, returns a new, altered `View`, that will replace the original `View` created. Built in `modifiers` such as `.bold()`, come built-in with Swift and all `Views` have access to these methods. `.bold()` can be used like so:

Reviewer 2: The first sentence on this line should be moved up with the text on line # 15. The other sentence should be moved below the \```pseudo code block, with the text on line # 33.

> 
> A `modifier` is a method that when called on a `View`, returns a new, altered `View`, that will replace the original `View` created. Built in `modifiers` such as `.bold()`, come built-in with Swift and all `Views` have access to these methods. `.bold()` can be used like so:

Reviewer: ```suggestion
```

> A `modifier` is a method that when called on a `View`, returns a new, altered `View`, that will replace the original `View` created. Built in `modifiers` such as `bold(_:)`, come built-in with Swift and all `Views` have access to these methods. `bold(_:)` can be used like so:

Reviewer 3: ```suggestion
A `modifier` is a method that when called on a `View`, returns a new, altered `View`, that will replace the original `View` created. Built in `modifiers` such as `.bold()`, come built-in with Swift and all `Views` have access to these methods. `.bold()` can be used like so:
```

> ```psuedo

Reviewer 2: ```suggestion
```pseudo
```

> struct MyView: View {
>     var body: some View {
>         VStack {
>             Text("Hello World")
>                 .bold()
>                 //Modifier methods like .bold() can be called underneath a text, button or even a View.
>         }
>     }
> }

Reviewer 2: ```suggestion
struct MyView: View {
  var body: some View {
      VStack {
          Text("Hello World")
              .bold()
              // Modifier methods like .bold() can be called underneath a text, button or even a View.
      }
  }
}
```

```

> Here the `.bold()` `modifier` is being called on the text: `Text("Hello World")`, which, replaces the original "Hello World" text with a new "**Hello World**" text. In this way the `modifier` method `.bold()` has been called on a view, and returned a new, altered view.

Reviewer 2: ```suggestion
Here, the `.bold()` modifier is called on the "Hello World" string in the `Text` view, which, replaces the original text with a bolded version of it. This ultimately returns a new, altered `Text` view.
```

> Here the `bold(_:)` `modifier` is being called on the text: `Text("Hello World")`, which, replaces the original "Hello World" text with a new "**Hello World**" text. In this way the `modifier` method `bold(_:)` has been called on a view, and returned a new, altered view.

Reviewer 3: ```suggestion
Here the `.bold()` `modifier` is being called on the text: `Text("Hello World")`, which, replaces the original "Hello World" text with a new "**Hello World**" text. In this way the `modifier` method `.bold()` has been called on a view, and returned a new, altered view.
```

## Example

> The following example displays a text, "Hello Sam", where nameText is the name of a user, and contains a button that increases `codingStreakCount` each time that the Button is pressed.

Reviewer 3: I think we'd prefer a simpler example just to demonstrate the use of modifiers in particular. And the introduction to the example should not just explain what the example does, but relate it to the subject of the entry.

Reviewer: @SSwiniarski I adjusted the example and image to match

cc: @AustinWheel

```swift
import SwiftUI

> struct ContentView: View {
>     var nameText: String = "Sam"
>     @State var codingStreakCount = 0
>     var body: some View {
>         VStack {
>             Text("Hello \(nameText)")
>                 .font(.title)
>                 .bold()
>                 .multilineTextAlignment(.center)
>             Button("Coding Streak: \(codingStreakCount)") {
>                 self.codingStreakCount += 1
>             }
>             .padding()
>             .overlay(
>                 RoundedRectangle(cornerRadius: 16)
>                     .stroke(.brown, lineWidth: 4))
>         }
>         .foregroundColor(.brown)
>         .padding()
>     }
> }

Reviewer 2: ```suggestion
struct ContentView: View {
  var nameText: String = "Sam"
  @State var codingStreakCount = 0
  var body: some View {
    VStack {
      Text("Hello \(nameText)")
        .font(.title)
        .bold()
        .multilineTextAlignment(.center)
      Button("Coding Streak: \(codingStreakCount)") {
        self.codingStreakCount += 1
      }
      .padding()
      .overlay(
        RoundedRectangle(cornerRadius: 16)
          .stroke(.brown, lineWidth: 4)
      )
    }
    .foregroundColor(.brown)
    .padding()
  }
}
```

Reviewer: @Dusch4593 does the 2-space indent apply to Swift as well? In Xcode, everything is automatically tabbed so it's better to copy/paste the examples here into their own project just to see.

```

> Each time a user presses the Button, the variable `codingStreakCount` will increase by one.

Reviewer 2: ```suggestion
Each time a user selects the `Button`, the value of the `codingStreakCount` variable will increase by one.
```

This code creates this output:

> ![alt text](https://github.com/AustinWheel/firstRep/blob/main/ExampleCode.png)

Reviewer: ```suggestion
![alt text](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-modifier.png)
```

> ![alt text](https://github.com/AustinWheel/firstRep/blob/main/ExampleCode.png)

Reviewer 3: Images should be uploaded to the docs/media folder, and any imbedded images need to link there. You also need to change the alt text of your example image to something descriptive.
